modules/presets/collection.js

```diff
@@ -93,81 +115,102 @@ export function presetCollection(collection) {
+                    if (dist + Math.min(value.length - alias.length, 0) < 3) {
+                        return { preset: a, alias: alias, fuzziness: dist };
+                    }
+                // only keep the one match with the lowest fuzziness
+                }).sort(function(a,b) { return a.fuzziness - b.fuzziness; })[0];
+            });
 
             // finds close matches to value in preset.terms
-            var similar_terms = searchable
-                .filter(function(a) {
-                    return (a.terms() || []).some(function(b) {
-                        return utilEditDistance(value, b) + Math.min(value.length - b.length, 0) < 3;
-                    });
-                });
+            var similar_terms = take(searchable, function(a) {
+                return utilArrayMapTruthy(a.terms(), function(term) {
+                    var dist = utilEditDistance(value, term);
+                    if (dist + Math.min(value.length - term.length, 0) < 3) {
+                        return { preset: a, term: term, fuzziness: dist };
+                    }
+                // only keep the one match with the lowest fuzziness
+                }).sort(function(a,b) { return a.fuzziness - b.fuzziness; })[0];
+            });
 
-            var similar_suggestions = suggestions
-                .map(function(a) {
-                    return { preset: a, dist: utilEditDistance(value, a.originalName.toLowerCase()) };
-                }).filter(function(a) {
-                    return a.dist + Math.min(value.length - a.preset.originalName.length, 0) < 1;
-                }).sort(function(a, b) {
-                    return a.dist - b.dist;
-                }).map(function(a) {
-                    return a.preset;
-                });
+            var similar_suggestions = take(suggestions, function(a) {
+                var dist = utilEditDistance(value, a.originalName);
+                if (dist + Math.min(value.length - a.originalName.length, 0) < 1) {
+                    return { preset: a, fuzziness: dist };
+                }
+            });
 
             var results = leading_name.concat(
+                leading_aliases,
                 leading_suggestions,
                 leading_terms,
                 leading_tag_values,
                 similar_name,
+                similar_aliases,
                 similar_suggestions,
                 similar_terms
             ).slice(0, maxSearchResults - 1);
 
             if (geometry) {
                 if (typeof geometry === 'string') {
-                    results.push(presets.fallback(geometry));
+                    results.push({ preset: presets.fallback(geometry) });
                 } else {
                     geometry.forEach(function(geom) {
-                        results.push(presets.fallback(geom));
+                        results.push({ preset: presets.fallback(geom) });
                     });
                 }
             }
```

Comment:
**Question**: If the desired `geometry` is already passed in this function as a parameter, why is the whole collection not filtered to only return presets that can be used with this geometry? Is this is a bug?